qiskit/mapper/_layout.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+A two-ways dict that represent a layout.
+
+Layout is the relation between logical (qu)bits and physical wires.
+Logical (qu)bits are tuples (eg, `('qr',2)`.
```

Comment:
Can you avoid using strings to refer to the first element of a wire tuple?
I know this PR is just creating the datastructure, and not enforcing the type that should go inside the two-ways dict.
But in #1189 and we have tried to steer clear of register names. So instead of `('qr', 2)`, it should be (qr, 2) where the first element is a QuantumRegister. This should be propagated throughout the PR, and in tests as well.

Reply:
Fixed in e47ba4a